packages/core/test/unit/application.unit.ts

```diff
@@ -35,9 +42,26 @@ describe('Application', () => {
 
   describe('component binding', () => {
     let app: Application;
+    const binding = new Binding('foo');
     class MyController {}
+    class MyClass {}
+    class MyProvider implements Provider<string> {
+      value() {
+        return 'my-str';
+      }
+    }
     class MyComponent implements Component {
       controllers = [MyController];
+      bindings = [binding];
+      classes = {'my-class': MyClass};
+      providers = {'my-provider': MyProvider};
+    }
+
+    class MyComponentWithDI implements Component {
```

Comment:
Please define this class inside the test that is using it (i.e. `it('binds from a component constructor'`) to avoid future temptation to add more stuff into this class.